build/android/src/net/minetest/minetest/MinetestAssetCopy.java

```diff
@@ -95,70 +97,75 @@ protected String doInBackground(String... files)
 			m_filenames    = new Vector<String>();
 			m_tocopy       = new Vector<String>();
 			m_asset_size_unknown = new Vector<String>();
-			String baseDir = 
-					Environment.getExternalStorageDirectory().getAbsolutePath()
-					+ "/";
-			
-			
+
+			Context con = getApplicationContext();
+			String baseDir = con.getFilesDir().getAbsolutePath() + "/";
+//			String baseDir =
+//					Environment.getExternalStorageDirectory().getAbsolutePath()
+//					+ "/";
+
+
 			// prepare temp folder
-			File TempFolder = new File(baseDir + "Minetest/tmp/");
-			
+			File TempFolder = new File(baseDir
+				+ MtNativeActivity.ProjectName + "/tmp/");
+
 			if (!TempFolder.exists())
 			{
 				TempFolder.mkdir();
 			}
 			else {
 				File[] todel = TempFolder.listFiles();
-				
+
 				for(int i=0; i < todel.length; i++)
 				{
 					Log.v("MinetestAssetCopy","deleting: " + todel[i].getAbsolutePath());
 					todel[i].delete();
 				}
 			}
-			
+
 			// add a .nomedia file
 			try {
-				OutputStream dst = new FileOutputStream(baseDir + "Minetest/.nomedia");
+				OutputStream dst = new FileOutputStream(baseDir
+					+ MtNativeActivity.ProjectName + "/.nomedia");
```

Comment:
I don't think this is necessary, the whole `/sdcard/Android` tree should be excluded from media indexing.